core/src/main/scala/core/builder/ServiceSpecValidator.scala

```diff
@@ -330,13 +332,53 @@ case class ServiceSpecValidator(
     service.unions.flatMap { union =>
       dupsError(
         s"Union[${union.name}] discriminator values",
-        union.types.map { unionType =>
-          unionType.discriminatorValue.getOrElse(unionType.`type`)
-        }
+        union.types.map(getDiscriminatorValue)
       )
     }
   }
 
+  private[this] def  getDiscriminatorValue(unionType: UnionType): String =
+    unionType.discriminatorValue.getOrElse(unionType.`type`)
+
+  private[this] def  getDiscriminatorKey(union: Union): String =
+    union.discriminator.getOrElse("discriminator")
```

Comment:
@mbryzek I could not find `"discriminator"` defined as a global variable or used anywhere else.

Is it already defined (and therefore should be used here)?

Reply:
Also, is this `getOrElse` even correct?

Reply:
per conversation - let's add 2 distinct error messgages and not hardcode `discriminator`